src/containers/Topbar/Topbar.js

```diff
@@ -12,6 +12,23 @@ import { themeConfig } from '../../config';
 
 const FormItem = Form.Item;
 const { Header } = Layout;
+const addressMaxDisplayLength = 11;
+const KEY_ADD_ADDRESS_BTN = 'add_address';
+
+function shortenAddress(text, maxLength) {
+  let ret = text;
+
+  const startLen = (maxLength - 3) / 2;
+  const endLen = (maxLength - 3) / 2;
+
+  console.log(`ret is ${ret}`);
+
+  if (ret.length > maxLength) {
+    ret = `${ret.substr(0, startLen)} ... ${ret.substr(ret.length - endLen)}`;
```

Comment:
the end part should be `${ret.substr(ret.length - endLen - 1)` right? since its based on 0 index?

Reply:
The first parameter of substr is start index, including itself
Let's say we have a string of 20 "01234567890123456789", maxLength is 11
endLen = (11-3) /2 = 4;
ret.subStr= (20 - 4) = ret.subStr(16);
So the end part will be the 16th,17th,18th and 19th, which are expected.